test/Makefile

```diff
@@ -0,0 +1,85 @@
+#
+# SYNOPSIS:
+#
+#   make [all]  - makes everything.
+#   make TARGET - makes the given target.
+#   make clean  - removes all files generated by make.
+
+# Points to the root of Google Test source
+GTEST_DIR = ../googletest/googletest
+
+# Points to the location of the Google Test libraries
+GTEST_LIB_DIR = .
+
+# The OpenTTD source
+USER_DIR = ../src
+
+# Flags passed to the preprocessor.
+# Set Google Test's header directory as a system directory, such that
+# the compiler doesn't generate warnings in Google Test headers.
+CPPFLAGS += -isystem $(GTEST_DIR)/include
+
+# Flags passed to the C++ compiler.
+CXXFLAGS += -g -Wall -Wextra -pthread -std=c++11
```

Comment:
Should be pulling these from the main makefile, or be generated with configure, imo